src/transformers/processing_utils.py

```diff
@@ -580,6 +584,23 @@ def get_processor_dict(
                     subfolder=subfolder,
                     _raise_exceptions_for_missing_entries=False,
                 )
+
+                # load chat template from a separate json if exists
+                # TODO @raushan: remove this hack after a few major releases
```

Comment:
Hmm I thought we could deprecate the `chat_template.json` when the old versions are very old to be in someone's env. I am okay with leaving `chat_template.json` but wouldn't that be different from what we have in LLMs? 

Another option I proposed was to add the template in processor's tokenizer. If we don't want to have the template in `processor.config`, this will be the better option IMO. WDYT?

Reply:
> Older versions of transformers will still break as processors won't accept the chat template in the config.


Oh, also, continuing on this. We can also discuss this internally on slack later. We recently added an option for some processors to accept any kwargs and I was hoping to start integrating new kwargs for VLMs. Does this comment mean that we can't do it and will need another hack to load those kwargs?

Reply:
> Hmm I thought we could deprecate the chat_template.json when the old versions are very old to be in someone's env

The problem is - when would this happen? There's some people who are still installing transformers v3, so we can't assume everyone will be working from the latest versions, even if it's a few months out. We can deprecate things within the code which we have full control over. Unfortunately, as the configs sit outside the library it means people can use them from the first version they're introduced until now. 

> Another option I proposed was to add the template in processor's tokenizer. If we don't want to have the template in processor.config, this will be the better option IMO. WDYT?

Just to be make sure we're talking about the same thing, this would mean adding into the tokenizer's config for the checkpoint e.g. [the tokenizer config here for a llava checkpoint](https://huggingface.co/llava-hf/llava-v1.6-mistral-7b-hf/tree/main)? 

It's a solution, but I don't think it semantically makes sense: the tokenizer is for processing text, whereas this introduces information about images.

> We recently added an option for some processors to accept any kwargs and I was hoping to start integrating new kwargs for VLMs. Does this comment mean that we can't do it and will need another hack to load those kwargs?

We definitely want this feature added for our processors, and gives us flexibility with future compatibility. It won't however fix things for older versions. My understanding is that if the chat_template gets added into the config, then older versions would still break when trying to load the config file.

Reply:
> It's a solution, but I don't think it semantically makes sense: the tokenizer is for processing text, whereas this introduces information about images.

Yes, I meant that checkpoint. Now I see why it's not in tokenizer's config, then this workaround of having a separate `json` for templates is indeed better. 

>  It won't however fix things for older versions. My understanding is that if the chat_template gets added into the config, then older versions would still break when trying to load the config file.

Yeah, this will be the problem in any case as we won't have compatibility with older transformers versions given how breaking are the new changes. Dealing with hub-transformers compatibility is harder than it looked like 😅  I guess for that feature of VLM processor refactor we'll have to wait a while and try to change slowly, to see users reactions...